src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs

```diff
@@ -98,5 +113,53 @@ public class RequestTokens : ITokenProvider {
 
             return _contentManager.GetItemMetadata(content).DisplayText;
         }
+
+        private static string FilterTokenParam(string token) {
+            string tokenPrefix;
+            int chainIndex, tokenLength;
+            if (token.IndexOf(":") == -1) {
+                return null;
+            }
+            tokenPrefix = token.Substring(0, token.IndexOf(":"));
+            if (!_textChainableTokens.Contains(tokenPrefix, StringComparer.OrdinalIgnoreCase)) {
+                return null;
+            }
+
+            // use ")." as chars combination to discover the end of the parameter
+            chainIndex = token.IndexOf(").") + 1;
+            tokenLength = (tokenPrefix + ":").Length;
+            if (chainIndex == 0) {// ")." has not be found
+                return Regex.Replace(token.Substring(tokenLength), @"[\(|\)]", "");
+            }
+            if (!token.StartsWith((tokenPrefix + ":"), StringComparison.OrdinalIgnoreCase) || chainIndex <= tokenLength) {
+                return null;
+            }
+            return Regex.Replace(token.Substring(tokenLength, chainIndex - tokenLength), @"[\(|\)]", "");
+        }
+        private static Tuple<string, string> FilterChainParam(string token) {
+            string tokenPrefix;
+            int chainIndex, tokenLength;
+
+            if (token.IndexOf(":") == -1) {
+                return null;
+            }
+            tokenPrefix = token.Substring(0, token.IndexOf(":"));
+            if (!_textChainableTokens.Contains(tokenPrefix, StringComparer.OrdinalIgnoreCase)) {
+                return null;
+            }
+
+            // use ")." as chars combination to discover the end of the parameter
+            chainIndex = token.IndexOf(").") + 1;
+            tokenLength = (tokenPrefix + ":").Length;
+            if (chainIndex == 0) { // ")." has not be found
+                return new Tuple<string, string>(Regex.Replace(token.Substring(tokenLength), @"[\(|\)]", ""), token.Length.ToString());
```

Comment:
Same. And also, if you reuse the same Regex then make one static and RegexOptions.Compiled.